linode/manager
# M3-5182: RQ-ify Images, Pt. 2

## Description 📝
Part 2 of the effort to transition Images from Redux to React Query.

At a high level, this PR:
- Allows for the removal of the `useImages` hook, `withImages.container.ts` (used to provide Images data from Redux to class components, and has been replaced by `images.container.ts` to provide Images data from RQ to class components), and Images from `useEntities` hook
- Adds a `useAllImagesQuery()` which is what we want in a lot of these cases, not the `useImagesQuery`
- Makes use of the `useAllImagesQuery()` across many files

## How to test 🧪
This PR touches some core functionality, namely:

1. `ImageSelect` fields (Linode Create flow, Linode Rebuild flow, StackScript Create flow)
2. Search (Images should show up for both normal and large accounts)
3. Migrate Linode modal

When comparing this branch to prod, it's important to ensure that anywhere that images are listed in the UI, you see the same options or results.

## Notes & Up Next
Regarding the [sorting issue that surfaced in Pt. 1](https://github.com/linode/manager/pull/8645#issuecomment-1359661600), there's an API ticket being worked on currently that will allow for sorting by Status. That will probably end up being rolled out before the RQ-ifying Images work is released to prod.

Part 3 will consist of addressing logic related to the Image upload functionality and final cleanup.

## To-Do:
- [x] Audit instances of `useImagesQuery` and `useAllImagesQuery` to ensure correct query is being used in each instance
- [x] Additional testing
